### ISSUE TITLE
UnitQuaternion quaternion number equality conflicts with rotation matrix equality

### ISSUE THREAD

**Issue by jenkspt** [JuliaGeometry/Rotations.jl#189]
```julia
julia> q = UnitQuaternion(1., 0., 0., 1.)
3×3 UnitQuaternion{Float64} with indices SOneTo(3)×SOneTo(3)(0.707107, 0.0, 0.0, 0.707107):
 0.0  -1.0  0.0
 1.0   0.0  0.0
 0.0   0.0  1.0

julia> -q
3×3 UnitQuaternion{Float64} with indices SOneTo(3)×SOneTo(3)(-0.707107, -0.0, -0.0, -0.707107):
 0.0  -1.0  0.0
 1.0   0.0  0.0
 0.0   0.0  1.0

julia> q ≈ -q
true
```

The problem here is that `q` and `-q` are represented by the same 3x3 rotation matrix, but in terms of quaternion math ops, `q` should not equal `q`.

I'm thinking that the quaternion math should take precedent here, and thus `q` should not equal `-q`

**Comment by hyrodium** [comment]
This will be fixed in #175.

```julia
julia> q = UnitQuaternion(1., 0., 0., 1.)  # with PR#175
3×3 UnitQuaternion{Float64} with indices SOneTo(3)×SOneTo(3)(0.707107, 0.0, 0.0, 0.707107):
 0.0  -1.0  0.0
 1.0   0.0  0.0
 0.0   0.0  1.0

julia> -q
3×3 StaticArrays.SMatrix{3, 3, Float64, 9} with indices SOneTo(3)×SOneTo(3):
 -0.0   1.0  -0.0
 -1.0  -0.0  -0.0
 -0.0  -0.0  -1.0
```

`UnitQuaternion` is a matrix rather than quaternion, so `-q` should be `SMatrix`.
This naming is confusing, so `UnitQuaternion` will be renamed `QuatRotation`. (https://github.com/JuliaGeometry/Rotations.jl/issues/171#issuecomment-950391326)

**Comment by hyrodium** [comment]
This issue has been solved by #175.
